### PR TITLE
distro: add rhel 84 support

### PR DESCRIPTION
**Pull request by jkozol**
Nightly repos are are added for testing.

The distro name for 84 is rhel-84 unlike prior rhel releases which fall under the umbrella name rhel-8. This fix will be generalized.

This runs on rhel 84 and can build a qcow2. But, there are many differences with the official image.

These differences are documented in #1013 

I am opening this as a draft PR to make it easier to collaborate on this with @chloenayon 

This PR is dependent on https://github.com/osbuild/osbuild/pull/518

### CODE REVIEW

File: cmd/osbuild-composer/composer.go [hunk 1]
```diff
@@ -107,6 +108,9 @@ func (c *Composer) InitWeldr(repoPaths []string, weldrListener, localWorkerListe
 
 	// TODO: refactor to be more generic
 	name := hostDistro.Name()
+	if name == "rhel-84" {
+		name = "rhel-8"
+	}
```
Comment (jkozol) [inline cmd/osbuild-composer/composer.go:113]: Will we eventually split up the rhel-8 distro into it's minor releases? If not should I make this more generic now?

**Comment by jkozol** [comment]
I can now generate a qcow2 image and boot it using a manifest built by this PR. But, for some reason, I am unable to get the image to build on my rhel 8.4 machine.

File: internal/distro/rhel84/distro.go [hunk 2]
```diff
@@ -0,0 +1,1107 @@
...
+		packages: []string{
+			"@core",
+			"chrony",
+			"dnf",
+			"dosfstools",
+			"efi-filesystem",
+			"efivar",
+			"efivar-libs",
+			"grub2-efi-x64",
+			"shim-x64",
```
Comment (jkozol) [inline internal/distro/rhel84/distro.go:824]: @gicmo I added these files to the qcow2 image type to follow what is supported in the official image. All x86_64 images will have grub2-efi-x64 and shim-x64 in their build packages. Do you have thoughts on which image types might need these packages installed? The official kickstarts do not explicitly install these packages.

**Comment by jkozol** [comment]
I am now able to successfully build rhel 8.4 images on rhel 8.4 so I am opening this up for review.

**Comment by atodorov** [comment]
> I am now able to successfully build rhel 8.4 images on rhel 8.4 so I am opening this up for review.

I don't see any changes to Jenkinsfile and I don't think schutzbot is running any jobs against the 8.4 definitions. If I'm wrong can you explain why ?

**Comment by jkozol** [comment]
> I don't see any changes to Jenkinsfile and I don't think schutzbot is running any jobs against the 8.4 definitions. If I'm wrong can you explain why ?

There is currently no test runner for 8.4. @major do you have any thoughts on what it will take to add 8.4 testing to schutzbot?

File: cmd/osbuild-composer/composer.go (hunk 1 above)
Comment (teg) [reply cmd/osbuild-composer/composer.go:115]: I think that would make sense. There are no immediate plans to split into the minor releases, that would anyway require more work as some customers would be on the generic RHEL8, but others would opt into a specific minor version.

File: schutzbot/deploy.sh [hunk 3]
```diff
@@ -51,3 +51,14 @@ fi
 greenprint "Installing the Image Builder packages"
 # Note: installing only -tests to catch missing dependencies
 retry sudo dnf -y install osbuild-composer-tests
+
+# Provision the software under test.
+/usr/libexec/osbuild-composer-test/provision.sh
```
Comment (teg) [inline schutzbot/deploy.sh:56]: How come this line is added (back?)?

File: schutzbot/deploy.sh [hunk 4]
```diff
@@ -51,3 +51,14 @@ fi
...
+
+# RHEL 8.4 still requires the nightly repos to be used
```
Comment (teg) [inline schutzbot/deploy.sh:58]: I think this chunk should be done in `provision.sh`, and `deploy.sh` should be kept generic. The thinking being that `deploy.sh` is specific to running these tests in schutzbot, but they should behave the same if you just execute them from the test RPM on a non-schutzbot system too.

**Review by teg — CHANGES_REQUESTED** [review]
Loving this!
One more thing I think we need: we must test that the hybrid images boot both using BIOS and UEFI. I know @gicmo had this set up, at some point, we just need to add the conditionals to the `qemu.sh` integration test.

File: schutzbot/deploy.sh (hunk 4 above)
Comment (jkozol) [reply schutzbot/deploy.sh:58]: So, each test should individually call `provision.sh` if they need it and `deploy.sh` wouldn't be needed for the tests to run unless they are in a schutzbot system? This makes sense to me.

**Comment by ondrejbudai** [comment]
This needs a rebase after #1063 lands.

**Comment by larskarlitski** [comment]
GitHub closed this pull request because I renamed osbuild's main branch of development to `main`, deleting `master`. If it is still relevant, please reopen it against the `main` branch. Thank you!

**Pull request closed without merging**